### ISSUE TITLE
`props => props.borderRadius` should not trigger `react/destructuring-assignment` 

### ISSUE THREAD

**Issue by dexteryy** [yannickcr/eslint-plugin-react#1637]
`props => <div id={props.id} />` should be considered to a stateless component, `props => props.borderRadius` should not

For example:

```js
import styled from 'react-emotion'

const ButtonWrapperDiv = styled.div`
  & .s-common-button {
    border-radius: ${props => props.borderRadius}px;
    background: ${props => props.backgroundColor};
    color: ${props => props.textColor};
    &:hover {
      background: ${props => props.hoverBackgroundColor};
    }
  }
```

BTW, it's better to provide an option to allow one line stateless component to use `props` directly.

For example:

```js
const ComonentA = props => <div id={props.id} />
```
  
or 

```js
function ComonentA(props) { return <div id={props.id} /> }
```

Otherwise it will be too complicated:

```js
const ComonentA = props => { const { id } = props; return <div id={id} /> }
```

**Comment by ljharb** [comment]
Use destructuring:
```js
function ComponentA({ id }) { return <div id={id} /> }
```

**Comment by ljharb** [comment]
Separately, what considers `props => props.borderRadius` an SFC? Can you elaborate?

**Comment by dexteryy** [comment]
<img width="841" alt="screenshot" src="https://user-images.githubusercontent.com/43011/34762630-64b7e8c0-f623-11e7-88d5-26f188265b02.png">

**Comment by ljharb** [comment]
aha, thank you :-)

This is definitely a bug, as that rule should be using our component detection, and [is not](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/destructuring-assignment.js).

**Comment by alexzherdev** [comment]
Looks like [it is now](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/destructuring-assignment.js#L28) but [it's not consulting the confidence level](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/destructuring-assignment.js#L102). These little components are being detected, but their confidence is 0.
I found that there is only one rule that makes use of the confidence value 🙁

**Comment by ljharb** [comment]
If we change the detection helper itself to not fire for a confidence level of zero, do any tests fail?

**Comment by ljharb** [comment]
That seems like it would be an improvement.
